### DISCUSSION TITLE
Some [Obsolete] RegisterProperty overloads

### DISCUSSION THREAD

**Discussion by michaelcsikos** [MarimerLLC/csla#2288] [Questions]
I'm having difficulty working out why some `RegisterProperty` methods are now deprecated,  while others are not.

```cs
[Obsolete]
RegisterProperty<P>(Expression<Func<T, object>> propertyLambdaExpression,
                    string friendlyName,
                    RelationshipTypes relationship)

// NOT [Obsolete]
RegisterProperty<P>(Expression<Func<T, object>> propertyLambdaExpression,
                    RelationshipTypes relationship)
```

I understand that `RelationshipTypes.Child` is no longer needed, but what about `RelationshipTypes.LazyLoad`? If we're using `LazyGetProperty` do we need to flag it?

**Comment by rockfordlhotka** [comment]
We only marked `Child` as obsolete because it has never been used within the CSLA codebase. I think we _planned_ on it having value, but chose a different implementation path.

The `LazyLoad` value is used in `BusinessBase` and `BusinessRule` to alter some behaviors due to you indicating that that property is lazy loaded.

The `PrivateField` value is used to alter behaviors as well.

**Reply by rockfordlhotka** [reply]
In terms of the `RegisterProperty` overloads, iirc we eliminated certain overloads that shouldn't be necessary. For example, you can (and should) use an attribute to indicate the friendly name, rather than hard-coding it into the `RegisterProperty` call.

**Reply by michaelcsikos** [reply]
OK. The thing with `[Display]` attributes is you have to constantly repeat yourself with the resource type for localised strings. I'm glad CSLA supports the `[DisplayName]` attribute which is not `sealed` and can make the attribute declarations so much neater.

**Reply by mikeclimbs23** [reply]
While we're on the subject of friendly name, can I get away with retrieving it using this code CO_States.ValueProperty.FriendlyName where co_states is Csla.BusinessBase or will it crash in some circumstances.

Thanks,

Mike

**Reply by michaelcsikos** [reply]
@mikeclimbs23 `FriendlyName` should never crash. If it hasn't been set via `RegisterProperty(...)`, `[Display]` or `[DisplayName]` it just returns the `PropertyInfo.Name`.
